hooks/pre-commit

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+from __future__ import print_function
+from future import standard_library
+standard_library.install_aliases()
+
+import commands
+import os
+import sys
+import urllib.request, urllib.parse, urllib.error
+from distutils.version import LooseVersion
+import re
+
+pattern = re.compile(r'script_version["\s:=]+(\d+\.\d+)')
+version_dict = {}
+versionfile = urllib.request.urlopen(
+    "https://raw.githubusercontent.com/henrykironde/retriever/1f18008838cecf7f8ac208d57cd2a578e50e1c2c/version.txt")
+versionfile.readline()
+
+for lines in versionfile:
+    key_values = lines.split(",")
+    version_dict[key_values[0]] = key_values[1]
+
+
+def get_script_version(file_name):
+    file_name = os.path.split(file_name)[1]
+    return LooseVersion(version_dict[file_name])
+
+
+# check if script directory changes
+status, output = commands.getstatusoutput("git diff --name-only origin/script2 scripts")
+
+# get list of changed scripts
+scripts_changed = output.split("\n")
+
+# get_version_text_files_from_master:
+# discus if we need the version files in master of in last commit
+# this is tricky
+version_files = subprocess.check_output("git diff --cached --name-only", shell = True).decode("UTF-8")
+
+versions_not_changed =[]  # if script changes we expect to change the version
+versions_updated = True
+for files in scripts_changed:
+    if files in version_files:
```

Comment:
I think we should change the logic here, If a script has changes, read in the file and use match = re.search() to get the version number.

update  the regex and make sure it gets the right version number 

mm is (a string having version number) the string below tries to cover all possible occurrences of version in both json and py files.


```
mm =  """
self.retriever_minimum_version = '2.0.dev'
self.version = '1.1.0'
    "retriever": "True",
    "retriever_minimum_version": "2.0.dev",
	"retriever_minimum_version": "2.0.0dev"
    "title": "Abalone Age and Size Data",
    "urls": {
        "data": "http://archive.ics.uci.edu/ml/machine-learning-databases/abalone/abalone.data"
    },
    "version": "1.2.1"
"""

```


```
>>> pattern = re.compile(r'[."]version["\s':=]+(\d+\.\d+.\d+)')
>>> match = re.search(pattern, mm)
>>> match.group(1)
'1.2.1'
```